crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -282,11 +282,7 @@ impl<'a> Resolver<'a> {
                         let id = self.interner.next_type_variable_id();
                         let typevar = Shared::new(TypeBinding::Unbound(id));
                         new_variables.push((id, typevar.clone()));
-
-                        // 'Named'Generic is a bit of a misnomer here, we want a type variable that
-                        // wont be bound over but this one has no name since we do not currently
-                        // require users to explicitly be generic over array lengths.
-                        Type::NamedGeneric(typevar, Rc::new("".into()))
+                        Type::TypeVariable(typevar)
```

Comment:
It turns out we do want to bind over these lengths, so the old comment was incorrect and was removed. We want to bind over array lengths when - for example - setting the return length equal to the length of an input.